include/bio/format/bcf_input_handler.hpp

```diff
@@ -0,0 +1,1364 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * brief Provides the seqan3::format_input_handler<seqan3::format_bcf> .
+ * \author Hannes Hauswedell <hannes.hauswedell AT decode.is>
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <ranges>
+#include <regex>
+#include <string>
+#include <string_view>
+#include <vector>
+
+// #include <seqan3/alphabet/views/char_strictly_to.hpp>
+#include <seqan3/alphabet/views/char_to.hpp>
+#include <seqan3/core/debug_stream.hpp> //TODO evaluate if there is a better solution
+#include <seqan3/core/debug_stream/detail/to_string.hpp>
+#include <seqan3/core/range/type_traits.hpp>
+#include <seqan3/utility/detail/to_little_endian.hpp>
+#include <seqan3/utility/type_list/traits.hpp>
+#include <seqan3/utility/views/to.hpp>
+
+#include <bio/detail/misc.hpp>
+#include <bio/detail/range.hpp>
+#include <bio/format/bcf.hpp>
+#include <bio/format/format_input_handler.hpp>
+#include <bio/stream/detail/fast_streambuf_iterator.hpp>
+#include <bio/var_io/dynamic_type.hpp>
+#include <bio/var_io/header.hpp>
+#include <bio/var_io/misc.hpp>
+#include <bio/var_io/reader_options.hpp> //TODO for field_types_raw; move somewhere else
+
+namespace bio::detail
+{
+//!\brief Low-level input iterator that points to the byte-region of a BCF record in the input file.
+class bcf_input_iterator
```

Comment:
shouldn't the input iterator get is own header file?

Reply:
Well, nothing else is expected to use it and it is detail...

Reply:
I don't mind putting it elsewhere to make the file smaller, but usually "is used by something that doesnt require the other thing" is the critierion that I use!